baybe/acquisition/utils.py

```diff
@@ -0,0 +1,43 @@
+"""Utilities for acquisition functions."""
+import warnings
+from typing import Union
+
+from baybe.acquisition import UpperConfidenceBound, qUpperConfidenceBound
+from baybe.acquisition.base import AcquisitionFunction
+from baybe.utils.basic import get_subclasses
+
+
+def str_to_acqf(name: str, /) -> AcquisitionFunction:
+    """Create an ACQF object from a given ACQF name."""
+    if name == "VarUCB":
```

Comment:
changed, depending on how precise you want the warning to be it would not be ass straighforward inc ase we deprecate other acqfs with parameters

Reply:
Ah 😄 that was actually not what I meant (though removing this other duplication is of course also nice). 

What I meant is: why do we need to raise the exact same warning both in the hook AND the converter? Perhaps I haven't fully thought it through, but my first guess would have been that raising the deprecation once in the correct place would suffice as long as we make sure all construction paths go through that piece of code!?

Reply:
ok sry for the confusion now I get it

one of the issues that appears is cyclical imports
my first attempts were not successful, might have to change the module structure to enable that

Reply:
ok @AdrianSosic this was done now by making `string_to_acqf` just use the global converter instead of repeatedly coding the logic (since the converter is exactly whats changed in acquisition.base where the second duplicated validation happened)
its in the last commit for you to check'n'resolve

Reply:
I think this seems like a reasonable approach 👍🏼 Glad that we could clean up this part 🥇 

From my side, everything ready for merge, will accept now. The PR has become quite long in the meantime. If there are an easy/obvious squashes, perhaps you can spend a minute to clean up. Otherwise, feel free to go 👍🏼